### PR TITLE
DM-11601 showChart API changes

### PR DESCRIPTION
**Pull request by tgoldina**
https://jira.lsstcorp.org/browse/DM-11601

- Renamed API function showPlot to showChart and removed fireflyData and fireflyLayout from its parameters.
- Firefly specific options are now under firefly property of data and layout, trace type can be fireflyScatter, fireflyHistogram, and fireflyHeatmap for Firefly extended charts.
- Updated addChart action creator to take care of this change and split plotly and firefly info on the fly.
- Added a workaround for hover text appearing without background in Firefox.

### CODE REVIEW

File: src/firefly/js/charts/ChartUtil.js [hunk 1]
```diff
@@ -288,7 +288,7 @@ export function getOptionsUI(chartId) {
     const {data, fireflyData, activeTrace=0} = getChartData(chartId);
     const type = get(data, [activeTrace, 'type'], 'scatter');
     const dataType = get(fireflyData, [activeTrace, 'dataType'], '');
-    if (isScatter2d(type)) {
+    if (dataType === 'fireflyScatter' || isScatter2d(type)) {
```
Comment (cwang2016) [inline src/firefly/js/charts/ChartUtil.js]: in here, either 'fireflyScatter' or scatter2d returns ScatterOptions. 
Question, for function getToolBarUI() next to this function, isn't 'fireflyScatter' treated the same as 'isScatter2d'? 
Also there is no 'selection box'  shown up for 'fireflyScatter' chart.

Comment (tgoldina) [reply src/firefly/js/charts/ChartUtil.js]: Thank you for finding this issue. Indeed, I made a change that for firefly enhanced charts plotly trace type is not assigned until the data are retrieved. This bug should be fixed in the latest commit.

**Comment by tgoldina** [comment]
Fixed the bug Cindy found, when scatter chart load resets table highlight to 0.

**Comment by cwang2016** [comment]
reviews completes. works well.

**Pull request merged**